pkg/shared/accountmgmtutil/ams.go

```diff
@@ -48,19 +65,64 @@ func GetUserSupportedInstanceTypes(ctx context.Context, spec remote.AmsConfig, c
 		return nil, err
 	}
 
-	var quotas []string
+	var quotas []QuotaSpec
 	for _, quota := range quotaCostGet.GetItems() {
-		quotaId := strings.TrimSpace(quota.GetQuotaId())
+		for _, quotaResource := range quota.GetRelatedResources() {
+			if quotaResource.GetResourceName() == spec.ResourceName {
+				if quotaResource.GetProduct() == spec.TrialProductQuotaID {
+					quotas = append(quotas, QuotaSpec{QuotaTrialType, 0, quotaResource.BillingModel})
+				} else if quotaResource.GetProduct() == spec.InstanceQuotaID {
+					remainingQuota := int(quota.GetAllowed() - quota.GetConsumed())
+					quotas = append(quotas, QuotaSpec{QuotaStandardType, remainingQuota, quotaResource.BillingModel})
+				}
+			}
+		}
+	}
+
+	return BattleOfInstanceBillingModels(quotas), err
+}
 
-		if quotaId == spec.TrialQuotaID {
-			quotas = append(quotas, QuotaTrialType)
+// This function selects the billing model that should be used
+// It represents some requirement to always use the same standard billing models
+// This function should not exist but it does represents some requirement that we cannot do on backend
+func BattleOfInstanceBillingModels(quotas []QuotaSpec) []QuotaSpec {
```

Comment:
Area of interest 2

Reply:
Billing models shouldn't be exposed to clients at all.

Reply:
We need to somehow abstract this properly on the backend.

Reply:
How is this used? Is it to determine what to send in the `plan` field when creating a new instance? The backend can already decide between `standard` and `eval` based on Quota. So technically it's only the size that we need from the UI. Unless of course we add other plans in the future that the user can choose from.